### ISSUE TITLE
Plots - legends hiding symbols

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/OSPSuite-R#1141]
With the default legend position (top right inside), legend background often covers some plotted area, leading to confusing perception of the figure. E.g. here

```
  simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
  sim <- loadSimulation(simFilePath)

  simData <- withr::with_tempdir({
    df <- dplyr::tibble(
      IndividualId = c(0, 0, 0),
      `Time [min]` = c(0, 2, 4),
      `Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood) [µmol/l]` = c(0, 4, 8)
    )
    readr::write_csv(df, "SimResults.csv")
    importResultsFromCSV(sim, "SimResults.csv")
  })

  obsData <- DataSet$new(name = "Observed")
  obsData$setValues(xValues = c(1, 3, 3.5, 4, 5), yValues = c(1.9, 6.1, 7, 8.2, 1))
  obsData$xUnit <- "min"
  obsData$yDimension <- ospDimensions$`Concentration (molar)`

  myDC <- DataCombined$new()
  myDC$addSimulationResults(simData, groups = "myGroup")
  myDC$addDataSets(obsData, groups = "myGroup")

  # Add second obs data
  obsData2 <- DataSet$new(name = "Observed 2")
  obsData2$setValues(xValues = c(0, 3, 4, 4.5, 5.5), yValues = c(2.9, 5.1, 3, 8.2, 1))
  obsData2$xUnit <- "min"
  obsData2$yDimension <- ospDimensions$`Concentration (molar)`
  myDC$addDataSets(obsData2, groups = "myGroup")

plotResidualsVsTime(myDC)
```
![image](https://user-images.githubusercontent.com/25903040/191706879-7dcdc31e-a610-4868-8b63-a820b64f9be8.png)

There is actually symbol at x = 4, y = 5. Can we somehow imrove the default behavior? Placing the legend outside the plot area is not something I like, because it may significantly reduce the available area. Can we make the background of the legend transparent?

And maybe also add a border around the legend?

**Comment by Felixmil** [comment]
Would this be acceptable ?

![image](https://user-images.githubusercontent.com/34234913/222694083-110ada55-a221-4791-be6d-f9e3155b426b.png)

**Comment by Felixmil** [comment]
All legend box available through TLF will now be accessible.

```r
myPlotConfiguration <- DefaultPlotConfiguration$new()

myPlotConfiguration$legendBackgroundColor <- "grey70"
myPlotConfiguration$legendBackgroundAlpha <- 0.3
myPlotConfiguration$legendBorderColor <- "grey30"
myPlotConfiguration$legendBorderType <- 3
myPlotConfiguration$legendBorderSize <- 1.12

plotResidualsVsTime(myDC, defaultPlotConfiguration = myPlotConfiguration)
```

yields:
![image](https://user-images.githubusercontent.com/34234913/222706521-07560884-754c-411d-a1ef-3c5471b04d4e.png)

**Comment by PavelBal** [comment]
@Yuri05 @msevestre @StephanSchaller @VanessaBa

Frame around legend by default - yes/no?

**Comment by VanessaBa** [comment]
Yes, please.

**Comment by Yuri05** [comment]
> Frame around legend by default - yes/no?

No.

**Comment by PavelBal** [comment]
@Felixmil No frame by default, but with (solid) frame in default `esqlabsR` configuration ;)

**Comment by Felixmil** [comment]
Implemented in #1216
